src/ReactTelephoneInput.js

```diff
@@ -110,6 +111,9 @@ var ReactTelephoneInput = React.createClass({
             this.props.onChange(this.state.formattedNumber, this.state.selectedCountry);
         }
     },
+    shouldComponentUpdate(nextProps) {
+        return !isEqual(nextProps, this.props);
```

Comment:
what happens when state is changed, but not props.? It will not update component ? Either it should be !isEqual(nextProps, this.props) || !isEqual(nextState, this.state). Please, can you include that condition?